src/cli/argument.rs

```diff
@@ -2,36 +2,103 @@ use clap::Arg;
 
 use super::validator;
 
+pub struct CliArgument<'a> {
+    long: &'a str,
+    short: Option<&'a str>,
+    help: &'a str,
+}
```

Comment:
You probably don't need the lifetime parameter here, you can just store `&'static` references directly since you only ever use `CliArgument<'static>` as far as I can tell.

```suggestion
pub struct CliArgument {
    long: &'static str,
    short: Option<&'static str>,
    help: &'static str,
}
```